include/libopencm3/stm32/mp1/ipcc.h

```diff
@@ -0,0 +1,156 @@
+/** @addtogroup ipcc_defines
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @cond */
+#ifndef LIBOPENCM3_IPCC_H
+/** @endcond */
+#define LIBOPENCM3_IPCC_H
+
+#include <libopencm3/stm32/memorymap.h>
+#include <libopencm3/cm3/common.h>
+
+/**@{*/
+enum ipcc_core {
+	IPCC_CPU_A7 = 1,
+	IPCC_MPU_M4 = 2
```

Comment:
I don't like these names, even in the ref manualy, it's MPU for the cortex-a, and MCU for the cortex-m.

Further, the entire IPCC ref manual just talks about cpu1 and cpu2, and I thin it's up to the user to decide which is for which, whereas you're basically trying to _assign_ how it is here.

Reply:
(Ive run out of time here, I'll get back to this later)

Reply:
> I don't like these names, even in the ref manualy, it's MPU for the cortex-a, and MCU for the cortex-m.
> 
> Further, the entire IPCC ref manual just talks about cpu1 and cpu2, and I thin it's up to the user to decide which is for which, whereas you're basically trying to _assign_ how it is here.

I probably took inspiration from what was proposed by st microelectronics examples. I don't mind changing them. Any advices ?